### ISSUE TITLE
tFileExcelSheetOutput : dissociate data location and style location

### ISSUE THREAD

**Issue by schwarz-ben** [jlolling/talendcomp_tFileExcel#24]
Hi,

    First of all I want to thank you for that great set of components that proved to be as useful as easy to handle.

  I come here with a suggestion that might provide some more versatility to the tFileExcelSheetOutput component when it comes to Excel styles. Recently I wrote a Talend job where I hade to write several consecutive blocks of lines in the same Excel-sheet, not knowing in advance the size of each block, and I though it would be great to be able to use something like alternate colors but with different colors... Or sometimes, simply to be able to repeat the style of line 1 throughout all lines of each block.
  This could be achieved for instance by providing another option "Use styles from another location" where we could select a sheet name and the (first) line that we want to consider in that sheet for copying styles.

**Comment by jlolling** [comment]
Good idea. I had actually already code prepared to allow to apply external styles to cells. The problem was Excel allows only a fixed limited number of styles and it was hard to detect if a style is already present.
I will keep thinking about that.

**Comment by schwarz-ben** [comment]
Hello, and thanks for your answer. Unfortunately, since I am far from being an Excel expert I am not sure I fully understand it :)
That being said, re-reading my suggestion I realize I was a bit messy. Let me try and rephrase my suggestion with an example :

Let's say I have an Excel file with a spreadsheet named "MyUgglyStyles" where I would have hard coded styles on a per-line basis . e.g. : 
- line-1 : no colors and arial bold in the first column, italic and blue background for the rest of the line
- line-2 : orange as a background and courrier sans serif on all columns

As I write a block of lines through tFileExcelSheetOutput, I would select a checkbox option "use existing styles from another line", and I would fill-up the subsequent parameters "sheet" and "line" respectively with "MyUgglyStyles" and 1 (or 2, depending on the style I want to apply).

Finaly, before saving my file I would then drop the MyUgglyStyles Sheet.

Now, I don't know how styles are handled by Excel. Are new styles created when your borrow some from other lines ?

**Comment by jlolling** [comment]
Hi Ben,

thats easy. In the tFileExcelWorkbookSave component you can setup a sheet to delete. Usually such sheets are template sheets and we do not want to have them in the final file.

Best regards
Jan

**Comment by jlolling** [comment]
Is your issue solved with my suggestions?

**Comment by jlolling** [comment]
In the last release the append rows mode is improved and does what you have suggested.

**Comment by jlolling** [comment]
Does the latest version of the component fulfill your requirements?
